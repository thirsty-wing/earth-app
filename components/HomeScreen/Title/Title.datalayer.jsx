import { useState } from "react";
import Title from "./Title";
import { yieldRotateGlobe } from "./Title.utils";


/**
 * Title of the home screen comprising a pressable which rotates a globe emoji
 *
 * @returns {JSX.Element}
 */
function TitleDatalayer() {

  const [globeOrientation, setGlobeOrientation] = useState(1);

  return(
    <Title
      globeOrientation={globeOrientation}
      onPress={yieldRotateGlobe({ globeOrientation, setGlobeOrientation })}
    />
  );
}

export default TitleDatalayer;
