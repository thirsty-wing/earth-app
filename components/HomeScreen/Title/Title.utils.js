/**
 * @callback setGlobeOrientation
 * @param {number}
 */

/**
 * yieldRotateGlobe.
 *
 * @param {number} globeOrientation
 * @param {function} setGlobeOrientation
 */
export function yieldRotateGlobe(globeOrientation, setGlobeOrientation) {
  return function() {
    const newGlobeOrientation = globeOrientation > 2
      ? 1
      : globeOrientation + 1;

      setGlobeOrientation(newGlobeOrientation);
  };
}
