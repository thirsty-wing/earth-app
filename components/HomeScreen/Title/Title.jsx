import { Text, Pressable } from "react-native";
import { theme } from "../../../utils/theme";
import { getGlobe } from "./Title.utils";


/**
 * Title of the home screen comprising a pressable which rotates a globe emoji
 *
 * @param {object} props
 * @param {1|2|3} [props.globeOrientation = 1]
 * @param {Function} [props.onPress]
 * @returns {JSX.Element}
 */
function Title(props) {

  const {
    globeOrientation = 1,
    onPress,
  } = props;

  const globe = getGlobe(globeOrientation);

  return(
    <Pressable
      style={{
        alignItems: "center",
        width: "100%",
      }}
      onPress={onPress}
    >
      <Text
        style={{
          color: theme.palette.text.primary,
          fontSize: theme.sizes.text.title,
          textAlign: "center",
        }}
      >
        What in the world? {globe}
      </Text>
    </Pressable>
  );
}

export default Title;
