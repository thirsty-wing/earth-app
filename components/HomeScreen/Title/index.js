import Title from "./Title.datalayer";

export default Title;
