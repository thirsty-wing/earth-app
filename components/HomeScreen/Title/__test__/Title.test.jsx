import Title from "../Title";
import { shallow } from "../../../../utils/test/shallow";


it("should render with globeOrientation as 1", () => {
  const content = shallow(<Title/>);

  expect(content).toMatchSnapshot();
});

it("should render with globeOrientation as 2", () => {
  const content = shallow(
    <Title
      globeOrientation={2}
    />
  );

  expect(content).toMatchSnapshot();
});

it("should render with globeOrientation as 3", () => {
  const content = shallow(
    <Title
      globeOrientation={3}
    />
  );

  expect(content).toMatchSnapshot();
});

