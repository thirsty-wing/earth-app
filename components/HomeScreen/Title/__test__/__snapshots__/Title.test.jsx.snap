// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with globeOrientation as 1 1`] = `
<Memo(Pressable)
  style={
    {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#EEF",
        "fontSize": 50,
        "textAlign": "center",
      }
    }
  >
    What in the world? 
    🌎
  </Text>
</Memo(Pressable)>
`;

exports[`should render with globeOrientation as 2 1`] = `
<Memo(Pressable)
  style={
    {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#EEF",
        "fontSize": 50,
        "textAlign": "center",
      }
    }
  >
    What in the world? 
    🌍
  </Text>
</Memo(Pressable)>
`;

exports[`should render with globeOrientation as 3 1`] = `
<Memo(Pressable)
  style={
    {
      "alignItems": "center",
      "width": "100%",
    }
  }
>
  <Text
    style={
      {
        "color": "#EEF",
        "fontSize": 50,
        "textAlign": "center",
      }
    }
  >
    What in the world? 
    🌏
  </Text>
</Memo(Pressable)>
`;
